File: 523-continuous-subarray-sum/continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        int csum= 0;
        HashMap<Integer,Integer> cSumMapMod = new HashMap<>();
        for(int i=0;i<nums.length;++i){
            csum= (csum+nums[i])%k;
            if(csum%k==0 && i>0) return true;
            else if(cSumMapMod.containsKey(csum)){
                if(cSumMapMod.get(csum)!=i-1) return true;
            }
            else cSumMapMod.put(csum,i);
        }
        return false;
    }
}

File: 523-continuous-subarray-sum/continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> ump;
        int csum=0;
        for(int i=0;i<(int)nums.size();++i){
            csum+= nums[i];
            if(csum%k==0 && i>0) return true;
            if(csum>k) csum%= k;
            if(ump.find(csum)!=ump.end()){
                if(ump[csum]!=i-1)  return true;
            }
            else ump[csum]= i;
        }
        return false;
    }
};